R/learner_randomPlantedForest_classif_rpf.R

```diff
@@ -0,0 +1,125 @@
+#' @title Classification Random Planted Forest Learner
+#' @author jemus42
+#' @name mlr_learners_classif.rpf
+#'
+#' @description
+#' Random Planted Forest: A directly interpretable tree ensemble.
+#'
+#' Calls [randomPlantedForest::rpf()] from 'randomPlantedForest'.
+#'
+#' @section Initial parameter values:
+#' - `loss`:
+#'   - Actual default: `"L2"`.
+#'   - Initial value: `"exponential"`.
+#'   - Reason for change: Using `"L2"` (or `"L1"`) loss does not guarantee predictions are valid
+#'     probabilities and more akin to the linear predictor of a GLM.
+
+#' @section Custom mlr3 parameters:
+#' - `max_interaction`:
+#'   - This hyperparameter can alternatively be set via `max_interaction_ratio` as
+#'     `max_interaction = max(ceiling(max_interaction_ratio * n_features), 1)`.
+#'     The parameter `max_interaction_limit` can optionally be set as an upper bound, such that
+#'     `max_interaction_ratio * min(n_features, max_interaction_limit)` is used instead.
+#'     This is analogous to `mtry.ratio` in [`classif.ranger`][mlr3learners::mlr_learners_classif.ranger], with
+#'     `max_interaction_limit` as an additional constraint.
+#'
+#' @templateVar id classif.rpf
+#' @template learner
+#'
+#' @section Installation:
+#' Package 'randomPlantedForest' is not on CRAN and has to be install from GitHub via
+#' `remotes::install_github("PlantedML/randomPlantedForest")`.
+#'
+#' @references
+#' `r format_bib("hiabu_2023")`
+#'
+#' @template seealso_learner
+#' @template example
+#' @export
+LearnerClassifRandomPlantedForest = R6Class("LearnerClassifRandomPlantedForest",
+  inherit = LearnerClassif,
+  public = list(
+    #' @description
+    #' Creates a new instance of this [R6][R6::R6Class] class.
+    initialize = function() {
+      param_set = ps(
+        max_interaction       = p_int(lower = 0, upper = Inf, default = 1, tags = "train"),
+        max_interaction_ratio = p_dbl(lower = 0, upper = 1, tags = "train"),
+        max_interaction_limit = p_int(lower = 1, upper = Inf, default = Inf, tags = "train", special_vals = list(Inf)),
```

Comment:
```suggestion
        max_interaction_limit = p_int(lower = 1, upper = Inf, tags = c("required", "train")),
```

Reply:
Oh I thought `Inf` has to be `special_vals`'d because it's not an integer, hence I got an error upon construction when I set `upper = Inf` ... at least I thought I did, now I can't reproduce it. 
Well, anyway, thanks for the fixes!